### PR TITLE
Generate Rust docs for the server's `OperationRegistry`

### PR DESCRIPTION
**Pull request by david-perez**
Including a quickstart example for the service being generated that the
user can copy and paste.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: codegen-server/src/main/kotlin/software/amazon/smithy/rust/codegen/server/smithy/ServerCargoDependency.kt [hunk 1]
```diff
@@ -21,6 +22,7 @@ object ServerCargoDependency {
     val Nom: CargoDependency = CargoDependency("nom", CratesIo("7"))
     val PinProjectLite: CargoDependency = CargoDependency("pin-project-lite", CratesIo("0.2"))
     val Tower: CargoDependency = CargoDependency("tower", CratesIo("0.4"))
+    val Tokio: CargoDependency = CargoDependency("tokio", CratesIo("1.0"), scope = DependencyScope.Dev)
```
Comment (crisidev) [inline codegen-server/src/main/kotlin/software/amazon/smithy/rust/codegen/server/smithy/ServerCargoDependency.kt:25]: Shouldn't we have both 

```
val Tokio: CargoDependency = CargoDependency("tokio", CratesIo("1.0"))
```

and 

```
val TokioDev: CargoDependency = CargoDependency("tokio", CratesIo("1.0"), scope = DependencyScope.Dev)
```

?

File: codegen-server/src/main/kotlin/software/amazon/smithy/rust/codegen/server/smithy/generators/ServerOperationRegistryGenerator.kt [hunk 2]
```diff
@@ -78,6 +88,83 @@ class ServerOperationRegistryGenerator(
...
+/// contains the `errors` member in your Smithy model), the function
+/// implementing the operation has to be fallible (i.e. return a
+/// [`Result`]). **You must register an implementation for all
+/// operations with the correct signature**, or your application
+/// will fail to compile.
+///
+/// The operation registry can be converted into an [`#{Router}`] for
+/// your service. This router will take care of routing HTTP
+/// requests to the matching operation implementation, adhering to
+/// your service's protocol and the [HTTP binding traits]  that you
```
Comment (crisidev) [inline codegen-server/src/main/kotlin/software/amazon/smithy/rust/codegen/server/smithy/generators/ServerOperationRegistryGenerator.kt:113]:
```suggestion
/// your service's protocol and the [HTTP binding traits] that you
```

File: codegen-server/src/main/kotlin/software/amazon/smithy/rust/codegen/server/smithy/generators/ServerOperationRegistryGenerator.kt [hunk 3]
```diff
@@ -78,6 +88,83 @@ class ServerOperationRegistryGenerator(
...
+/// used in your Smithy model. This router can be converted into a
+/// type implementing [`tower::make::MakeService`], a _service
+/// factory_. You can feed this value to a [Hyper server], and the
```
Comment (crisidev) [inline codegen-server/src/main/kotlin/software/amazon/smithy/rust/codegen/server/smithy/generators/ServerOperationRegistryGenerator.kt:116]:
```suggestion
/// type implementing the _service factory_ [`tower::make::MakeService`].
/// You can feed this value to a [Hyper server], and the
```

**Review by crisidev — APPROVED** [review]
LGTM. Minor comments.

**Comment by david-perez** [comment]
What this looks like:

image

**Pull request merged**